src/backend/core/models.py

```diff
@@ -541,48 +561,124 @@ def delete_version(self, version_id):
             Bucket=default_storage.bucket_name, Key=self.file_key, VersionId=version_id
         )
 
+    def get_nb_accesses_cache_key(self):
+        """Generate a unique cache key for each document."""
+        return f"document_{self.id!s}_nb_accesses"
+
+    @property
+    def nb_accesses(self):
+        """Calculate the number of accesses."""
+        cache_key = self.get_nb_accesses_cache_key()
+        nb_accesses = cache.get(cache_key)
+
+        if nb_accesses is None:
+            nb_accesses = DocumentAccess.objects.filter(
+                document__path=Left(models.Value(self.path), Length("document__path")),
+            ).count()
+            cache.set(cache_key, nb_accesses, timeout=3600)
```

Comment:
What about managing this timeout with a setting ? It's still faster to change it in production if needed than releasing a new version

Reply:
Defining timeout is already handled in Django's CACHES setting. I'll add a setting to configure its TIMEOUT and remove it from here.